packages/sdk/src/utils/monitoring.ts

```diff
@@ -0,0 +1,207 @@
+import type { EventMonitoringConfig } from '@moonbeam-network/xcm-builder';
+import type { AssetRoute } from '@moonbeam-network/xcm-config';
+import { getPolkadotApi } from '@moonbeam-network/xcm-utils';
+import type { ApiPromise } from '@polkadot/api';
+import type { EventRecord } from '@polkadot/types/interfaces';
+import type { ISubmittableResult } from '@polkadot/types/types';
+
+interface ListenToDestinationEventsProps {
+  route: AssetRoute;
+  monitoringConfig: EventMonitoringConfig;
+  messageId?: string;
+  onDestinationFinalized?: () => void;
+  onDestinationError?: (error: Error) => void;
+}
+
+const ENABLE_LOGGING = true;
```

Comment:
Not sure about that. It will spam in dapps right? Then we need to allow to overwrite etc. Or logger levels.

Reply:
this is a mistake actually, I was testing and must have committed accidentally